### PR TITLE
Sec Rifle Port

### PR DESCRIPTION
**Pull request by Fox-McCloud**
Ports over TG's sec rifles and https://github.com/tgstation/-tg-station/pull/13891

Also ports over: https://github.com/tgstation/-tg-station/pull/10019

securer crates: https://github.com/tgstation/-tg-station/pull/813

The sec rifles are basically a 20 shot ballistic rifle with a slightly lower than average firing rate; these cannot be made in R&D and are a "cargo only" item. That said, the ammunition for them can be produced by R&D at the start of the round (with no research required to be done).

Also increases the buck and rubbershot pellets by 1

Combat shotgun:
- The combat shotgun will auto-pump after each shot
- Ammo reduced from 8 to 6.

Secure Crates
- Crates now have 1000 health
 - RIP guncargo.

Fixes
- Fixes some pathing on ammo casings and projectiles that was making them shoot the incorrect bullet

### CODE REVIEW

**Comment by DaveTheHeadcrab** [comment]
Seems a bit redundant with the Saber SMG already being a thing and being able to load very similar ammo types.

**Comment by Regen1** [comment]
While it's nice to see some new weapons here and there, this can really fuck over some antags, ballistic weapons are SO MUCH STRONGER than lasers because of the chance of breaking bones and bullet embed.
Especially single antags such as wizards, and "all out" antags such as nuke ops, this is a big nerf to them.

**Comment by Fox-McCloud** [comment]
>Especially single antags such as wizards, and "all out" antags such as nuke ops, this is a big nerf to them.

We've had combat shotguns and riot shotguns in for quiteeee some time now. Yes, they have a lower firing rate than these, but they gobs more damage and way more accessible ammunition.

**Comment by Earthdivine** [comment]
>We have had Shotguns

Except Shotguns also fire way less precise. This would be the equivalent of giving every officer a revolver in terms of stunning/bone breaking power.

:+1: for me, because yes, Nuke Ops will be screwed by this because it wont be stopped by a E-Shield. But that's having to have Cargo order them, wait for them, then get them to someone to open them. By then Nuke Ops generally have done their job, or intercepted the guns. 

Wizards....have way more things to worry about then this. coughchemistrycough.

Also I don't see what access level these are at. Are they Warden and higher or Sec Officer?

**Comment by Fox-McCloud** [comment]
>Except Shotguns also fire way less precise. This would be the equivalent of giving every officer a revolver in terms of stunning/bone breaking power.

This only goes for pellets, actually--for regular slugs, they're just as accurate as a revolver.

As for the level; it requires armory level access to open them.

**Comment by Regen1** [comment]
Shotguns are also pump action, but I suppose we'll see how these work out

I am not against the PR, but not super excited for it.
We need an "indifferent" thumb picture

**Comment by Robertha89** [comment]
wait, isnt this the gun that caused a huge drama regarding it, because it was op as hell?

**Comment by Spacemanspark** [comment]
Yes, @robertha89

**Comment by Robertha89** [comment]
Okay, I've used those guns, and they wreck so hard that it's not even funny, especially on this server . Let's list the reason why.
* Essentially duplicate code, we got something that does the same, but is probably better
* Bullet break bones
* Not to forget about decapitation
* Super easy to get, super cheap to make, and easy to distribute, science at least got lockboxes, cargo got crates they can break open at round start.
* Science still needs to research it
* Science also needs help from mining to even build it
* Imagine if energy weapons had power cells you could replace, again.

Projectile weapons are limited, due to how powerful they are, you can drag around enough ammo to kill entire departments, bullets are not that visible, and thus harder to dodge, thus you can easily be hit 3-5 times, and then you can't get away because your head is broken, and you already got negative health from your ribs.

There is enough of a problem with the saber as it is, no reason to add a second cheaper and easier to get one.

I mean, I would love to have this, but the balance would be so wrecked, especially for wizard thats been nerfed so hard only experience players who played wizard several times from before can even succeed.

I like the idea, but this is a bad one as far as ideas go, unless the breaking of bones and decapitation system is re-worked.

**Comment by Plotron** [comment]
I always thought that Security is using energy based weaponry mostly for their versatility (ubiquitous rechargers, no structural damage, shoot through windows) and easily reversible injuries (when an arrest goes wrong, you don't end up with a dozen of broken bones). Ballistic weapons would be preferred by offensive/attack forces (magazines, crippling/stopping power) and energy weapons by peacekeeping/security/defense forces (ammo is as cheap as power; wall outlets; good enough to scare hostiles away, but not permanently cripple them - good for arrests and negotiations)

This rifle goes against this notion. Such a weapon, if @Robertha89's feelings are to be trusted, should be restricted.

**Comment by Fox-McCloud** [comment]
Guns haven't decapitated in over 6+ months; that's flat out wrong.

Some facts about these guns before it gets lost amongst opinion:

- They have a 20 round magazine
- their bullets do 20 damage per shot
- they have an ever so slightly slower firing speed than laser guns
- They do not have burst fire
- They cannot be suppressed
- It has much lower DPS than the Riot Shotguns that security *starts* out with (yes, this factors in the time it takes to pump them)--it takes about 9 seconds to deal ~360 damage with the WT's vs the Riot Shotgun's' ~6-7 seconds

**Comment by Regen1** [comment]
We can run around passing numbers and all, but the fact remains that ballistic weapons are fundementally much stronger than energy weapons when it comes to straight out killing power, which is not what security is supposed to do unless something goes very wrong.

From what combat shotguns show, these will either end up in the hands of a greedy captain / HoS / blueshield, or more likely, the entire cargo team

**Comment by Fox-McCloud** [comment]
>which is not what security is supposed to do unless something goes very wrong.

Which is why they're locked away in a cargo crate, much like combat shotguns are (this is incidentally where the drama from these rifles came from; the fact they were pre-placed on the map with spare ammo and not their existence).

>From what combat shotguns show, these will either end up in the hands of a greedy captain / HoS / blueshield, or more likely, the entire cargo team

This isn't what happens on the live server though; Riot Shotguns are never used, and combat shotguns are almost never purchased by anyone, even cargo's own team.

**Comment by HereticChurch** [comment]
I always liked the idea of these. :+1: 
If we add them, are we going to be buffing the Saber as well, seeing how similar they are?

**Comment by Fox-McCloud** [comment]
@HereticChurch 

The Saber has 30 round mag capacity, the ability to be silenced, and burst fire mode, which definitively sets it apart from this rifle.

**Comment by Fox-McCloud** [comment]
Updated it and ported over another small gun-related PR; this one relates to the shotgun; see OP.

**Comment by Plotron** [comment]
In this case, I'd like to be nitpicky about the description of the gun.
Automatic? Semi-automatic at best according to what you've said, @Fox-McCloud, if it has worse fire rate than our laser guns. It's not an ARG or Saber.
In order not to confuse players, it'd be better to call it *semi-auto*.
Also, may I propose that it'd be called *obsolete* instead of *outdated*? I mean it sure is obsolete after the switch to energy-based weaponry in security forces and *obsolete* tells more than *outdated* does.
Sure, it's not nearly as deadly as actual military gear (Nukies, ERT specials), but it can be attributed to its role. I just think it sounds better this way.  http://dictionary.reference.com/browse/obsolete

Interesting fact about the ammo used: https://en.wikipedia.org/wiki/HK_4.6%C3%9730mm
Does this gun really have what it takes to be called a PDW, though?

File: code/modules/projectiles/ammunition/ammo_casings.dm [hunk 1]
```diff
@@ -28,17 +28,37 @@
 /obj/item/ammo_casing/c9mmap
 	desc = "A 9mm bullet casing."
 	caliber = "9mm"
-	projectile_type =/obj/item/projectile/bullet/armourpiercing
+	projectile_type = "/obj/item/projectile/bullet/armourpiercing"
```
Comment (TheDZD) [inline code/modules/projectiles/ammunition/ammo_casings.dm]: Bullets still use text paths? That is absolutely horrid.

Comment (Fox-McCloud) [inline code/modules/projectiles/ammunition/ammo_casings.dm]: Yyyyyup; it's horrible.

I figured it was just bad practice--but nope; the Saber SMG was firing normal bullets for all its magazine types without it...

File: code/modules/projectiles/guns/projectile/automatic.dm [hunk 2]
```diff
@@ -102,14 +102,30 @@
 	icon_state = "c20r[magazine ? "-[Ceiling(get_ammo(0)/4)*4]" : ""][chambered ? "" : "-e"][silenced ? "-suppressed" : ""]"
 	return
 
+/obj/item/weapon/gun/projectile/automatic/wt550
+	name = "security auto rifle"
+	desc = "An outdated personal defence weapon. Uses 4.6x30mm rounds and is designated the WT-550 Automatic Rifle."
```
Comment (TheDZD) [inline code/modules/projectiles/guns/projectile/automatic.dm]: "is designated the WT-550 Automatic Rifle" sounds a bit odd, desginated seems kind of out of place/the clause is structured a bit oddly.

Comment (Spacemanspark) [inline code/modules/projectiles/guns/projectile/automatic.dm]: "An outdated personal defense weapon, the WT-550 Automatic Rifle fires 4.6x30mm rounds, and was typically used in law enforcement."

File: code/modules/projectiles/ammunition/ammo_casings.dm (hunk 1 above)
Comment (Tastyfish) [inline code/modules/projectiles/ammunition/ammo_casings.dm]: I can't imagine what the person was thinking when they did that.

Comment (Fox-McCloud) [inline code/modules/projectiles/ammunition/ammo_casings.dm]: It's amazing how many things were textpathed in the old days--not a clue why.

**Comment by TheDZD** [comment]
I only have two major concerns with the weapon.
(1) If security gets a hold of it, despite it firing slower than a laser gun, it has an incredibly large clip size, and extra magazines can be carried around. Having this much ammo allows for sec to draw out battles against antags that would in almost all other situations win a war of attrition (blob in particular comes to mind). Combat/riot shotguns aren't as bad when it comes to this due to the fact that they have limited clip sizes.
(2) Gun cargo. Not really much else to say here other than the fact that these are a lot more efficient ammo-wise than the other weapons that cargo has access to. It's especially bad given how trivial it is to break open crates. At least with the Saber SMG, you either need an emag, or armory access.

The first point is countered to some extent by the fact that Saber SMGs are a thing, and it does actually surprise me that those aren't used more often.

**Comment by Fox-McCloud** [comment]
addressed concerns about breaking crates open; should be muchhhh harder and far more time consuming+resource intensive to do so, now.

File: code/modules/projectiles/ammunition/ammo_casings.dm (hunk 1 above)
Comment (tigercat2000) [inline code/modules/projectiles/ammunition/ammo_casings.dm]: probably didn't realize you can store actual paths in variables
mainly because that's insane

**Comment by tigercat2000** [comment]
please don't make the secure crates _all_ difficult to open
it's the only thing that seperates them from bullshit lockboxes, is the ability to grab an emitter and bust them open

**Comment by Plotron** [comment]
I agree with @tigercat2000, but that's probably because I'm a fan of the *cargo scheme*. It's one of the less violent ways in which you can get shit you want (but people refuse to give) without having to steal IDs, rob people or break into places and steal shit other people really need.

**Comment by Finalsong** [comment]
I personally like Bay's ballistics more than /tg/'s but hey this is cool :+1:

**Pull request merged**